ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -627,18 +617,52 @@ public CompactorMR getMrCompactor() {
     return new CompactorMR();
   }
 
-  private void markFailed(CompactionInfo ci, String errorMessage) {
-    if (ci != null && StringUtils.isNotBlank(errorMessage)) {
-      ci.errorMessage = errorMessage;
+  private void markWithState(CompactionInfo ci, Object error, CompactionState state) {
+    if (ci == null) {
+      LOG.warn("Could not set compaction info state because it is null!");
+      throw new RuntimeException("Could not set compaction info state because it is null!");
     }
+    String defaultMessage = "Error during compaction. Compaction info: " + ci;
```

Comment:
this could be extracted into the getErrorDetails method and reused under the switch statement

Reply:
I created a logError(CompactionInfo ci, Object error, CompactionState state) method instead, I think it's even better in case of avoiding code duplication.